tailwind/ui/Checkbox.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from "react"
+import { RxCheck } from "react-icons/rx"
+import * as CheckboxPrimitive from "@radix-ui/react-checkbox"
+
+import { cn } from "@/lib/utils/cn"
+
+export type CheckboxProps = React.ComponentPropsWithoutRef<
+  typeof CheckboxPrimitive.Root
+>
+
+const Checkbox = React.forwardRef<
```

Comment:
@pettinarip this will be different than Chakra, where the preferred convention is to work with [Form Components](https://ui.shadcn.com/docs/components/form) to set the label, controlled values, and layout.

ShadCN like `react-hook-form` for this, and Radix has their own [Form component in preview](https://www.radix-ui.com/primitives/docs/components/form). There is also still an [Open RFC PR](https://github.com/radix-ui/primitives/pull/1977)

Or, modify this to reflect how Chakra sets up rendering the `label`. Would have to do the same with `switch` and `radio`. Personally, I'd rather have more control with the layout of the label and/or utilizing a higher-level form component as mentioned above.

Reply:
Update: A form component library of some kind will need to be used, as the form controls do not render their own `input`.

Checkbox and radio only render a `button` which is the control itself. Using `FormControl` from ShadCN, say, and wrap it around the `Checkbox` will render the checkbox button and a hidden `input`.

Reply:
Makes sense. I would lean towards using `react-hook-form` since it is the recommended approach and we have used it on other projects.

But it shouldn't be a must to wrap every checkbox with a form container, right? for example the checkbox used in https://ethereum.org/en/wallets/find-wallet/ or do you see a problem using the button tag instead of the input?

As for the scope of this PR, we should be good as it is, right?

Reply:
With the checkbox in the find wallet page, it should be possible as it stands, as there is still the `onChange` prop, which is all that is really needed there for functionality.

So in Chakra, the `Checkbox` is already rendering a hidden `input` and the control is a styled `span` tag.


It's essentially built as (in the primitive HTML structure)
```tsx
<label>
  <input hidden />
  <span /> // The visible control
  <span>{ children }</span> // This renders if there is children text
</label>
```

But we can be more flexible here. So leaving the ShadCN checkbox as is will be fine.

ShadCN checkbox is more primitive than Chakra's, because it's just the control as a `button`. The `button` is used so that the control accessible and easily click to toggle `checked` or `unchecked`. You would then do a controlled hook-up to a hidden `input` when used in a form. 

In short, I think we can leave it as is.

Reply:
Got it. Cool, then we are fine. Thanks for the explanation.